### PR TITLE
add BLE Qingping CGDK2

### PR DESCRIPTION
**Pull request by Flobul**
## Description:

Modifications on xsns_62_esp32_mi_ble.ino to add BLE device : Qingping CGDK2 

## Checklist:
  - [X] The pull request is done against the latest development branch
  - [X] Only relevant files were touched
  - [X] Only one feature/fix was added per PR and the code change compiles without warnings
  - [(not concerned)] The code change is tested and works with Tasmota core ESP8266 V.2.7.4.9
  - [X] The code change is tested and works with Tasmota core ESP32 V.2.0.6
  - [X] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

### CODE REVIEW

**Comment by Flobul** [comment]
Integration was discussed here : https://github.com/arendst/Tasmota/discussions/17370

**Comment by Jason2866** [comment]
@Staars  Good to merge?

**Comment by Staars** [comment]
Looks good to me.

File: tasmota/tasmota_xsns_sensor/xsns_62_esp32_mi_ble.ino [hunk 1]
```diff
@@ -1073,10 +1089,12 @@ int MI32advertismentCallback(BLE_ESP32::ble_advertisment_t *pStruct)
     TasAutoMutex localmutex(&slotmutex, "Mi32AdCB2");
     switch(UUID){
       case 0xfe95: // std MI?
-      case 0xfdcd: // CGD1?
       {
         MI32ParseResponse(ServiceData, ServiceDataLength, addr, RSSI);
       } break;
+      case 0xfdcd: { // CGD1?
+        MI32ParseCGDK2Packet(ServiceData, ServiceDataLength, addr, RSSI);
+      } break;
```
Comment (barbudor) [inline tasmota/tasmota_xsns_sensor/xsns_62_esp32_mi_ble.ino:1097]: Doesn't that break the compatibility with CDG1 ?

Comment (Staars) [reply tasmota/tasmota_xsns_sensor/xsns_62_esp32_mi_ble.ino:1097]: I am afraid you are right.

Comment (Flobul) [reply tasmota/tasmota_xsns_sensor/xsns_62_esp32_mi_ble.ino:1097]: Oh yeah, maybe. :(
This should patch it: 

```     
      case 0xfe95: // std MI?
      case 0xfdcd: // CGD1 & CGDK2
      {
        if (ServiceDataLength == 17){ // CGDK2
          MI32ParseCGDK2Packet(ServiceData, ServiceDataLength, addr, RSSI);
        } else {
          MI32ParseResponse(ServiceData, ServiceDataLength, addr, RSSI);
        }
      } break;
```

Comment (meloman27) [reply tasmota/tasmota_xsns_sensor/xsns_62_esp32_mi_ble.ino]: Now the CGD1 device does not work correctly, it constantly changes from CGD1 to CGDK2.

**Pull request merged**